### PR TITLE
feature/acceleration_and_deceleration

### PR DESCRIPTION
**Pull request by raulgonzalezupc**
- Acceleartion of the player when starts moving
- Deceleration of the player when stops moving
- Inspector on the variables

### CODE REVIEW

File: Scripts/src/Script/PlayerMovement.cpp [hunk 1]
```diff
@@ -298,24 +300,52 @@ void PlayerMovement::ApplyVelocity()
...
+	{
+		return current_speed + dtime;
+	}
+	if (difference < -dtime)
+	{
+		return current_speed - dtime;
+	}
+	return final_speed;
+}float PlayerMovement::ComputeSpeedWithDeAcceleration(float final_speed, float current_speed, float dtime)
+{
```
Comment (marcorod94) [inline Scripts/src/Script/PlayerMovement.cpp:337]: Codestyle

**Review by marcorod94 — COMMENTED** [review]
Looks nice great job

**Review by marcorod94 — APPROVED** [review]
Nice job

File: Scripts/src/Script/PlayerMovement.cpp (hunk 1 above)
Comment (raulgonzalezupc) [reply Scripts/src/Script/PlayerMovement.cpp]: didn't notice, sorry, fixed!

**Review by Dartemiss — CHANGES_REQUESTED** [review]
The feeling for me is very weird is like skating on ice, especially when you change direction or you spin.
I would apply acceleration only the first time you are putting input, and if u change the input (turn or spin velocity) don't apply it.

File: Scripts/src/Script/PlayerMovement.h [hunk 2]
```diff
@@ -96,7 +98,9 @@ class PlayerMovement : public Script
 
 	float3 direction;
 	float3 velocity;
-
+	float acceleration = 0.035F;
+	float deceleration = 0.035F;
```
Comment (Dartemiss) [inline Scripts/src/Script/PlayerMovement.h:102]: f lowercase for code style

**Review by mariofv — COMMENTED** [review]
I tried it, and I have the feeling that the player is sliding when turning :S

**Comment by raulgonzalezupc** [comment]
> I tried it, and I have the feeling that the player is sliding when turning :S
This is probably because you are turning when you haven't reach yet the top velocity and is still doing the acceleration. I'm going to change it to make it smooth

**Comment by raulgonzalezupc** [comment]
> I tried it, and I have the feeling that the player is sliding when turning :S

This is probably because you are turning when you haven't reach yet the top velocity and is still doing the acceleration. I'm going to change it to make it smooth

**Review by Dartemiss — APPROVED** [review]
LGTM

**Pull request merged**